fix(transformer/react): isStaticChildren should be false when there is only one child

Dunqing opened the pull request:
We found a warning report [here](https://github.com/facebook/react/blob/206df66e70652e85711c3177ce1a0459609a7771/packages/react/src/jsx/ReactJSXElement.js#L614-L632). It caused by we transform incorrectly

Dunqing commented:
* **#5745** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/5745?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Dunqing and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

Boshen commented on tasks/transform_conformance/tests/babel-plugin-transform-react-jsx/test/fixtures/static-children/input.jsx:
```diff
@@ -0,0 +1 @@
+<div><div></div></div>
```
Can you add some comments here? Otherwise this is a really plain confusing test.

Dunqing replied:
Added

Dunqing committed:
fix(transformer/react): isStaticChildren should be false when there is only one child (#5745)

We found a warning report [here](https://github.com/facebook/react/blob/206df66e70652e85711c3177ce1a0459609a7771/packages/react/src/jsx/ReactJSXElement.js#L614-L632). It caused by we transform incorrectly
